### DISCUSSION TITLE
Is there a fish equivalent to bash's quick substitution (^)?

### DISCUSSION THREAD

**Discussion by tmercswims** [fish-shell/fish-shell#10018] [Q&A] [answered]
In bash you can use the caret (`^`) to rerun the previous command with some substitution. For example:
```bash
$ sudo apt update
# ...
$ ^date^grade
# (runs `sudo apt upgrade`)
```

Is there an equivalent (or partial equivalent) to this construct in fish? (Or a plan for such a thing, I suppose.)

**Comment by faho** [comment]
> Or a plan for such a thing, I suppose.)

There is not. Philosophically, we don't want to add obscure shortcuts like that.

You can, however, write it yourself as an [abbreviation](https://fishshell.com/docs/current/cmds/abbr.html).

See e.g. [my abbrs.fish](https://github.com/faho/fish-snippets/blob/main/conf.d/abbrs.fish) as inspiration.

**Reply by faho** [reply, marked as answer]
I had tried this out myself a while back:

```fish
abbr histreplace --regex '\^.*\^.*' --function replace_history --position anywhere
function replace_history
    set -l kv (string split '^' -- $argv[1])
    string replace -- $kv[2] $kv[3] $history[1]
end
```

**Reply by tmercswims** [reply]
Oh that's brilliant! I haven't explored regex abbreviations much yet, so it didn't come to mind to try that out, haha. Thank you!
